gobblin-compaction/src/main/java/org/apache/gobblin/compaction/suite/CompactionSuiteBase.java

```diff
@@ -130,7 +127,6 @@ public String datasetURN() {
    * @return a map-reduce job which will compact files against {@link org.apache.gobblin.dataset.Dataset}
    */
   public Job createJob (FileSystemDataset dataset) throws IOException {
-    configurator = CompactionJobConfigurator.instantiateConfigurator(this.state);
```

Comment:
Seems the configurator has to be instantiated lazily. Check `optionalInit` method which could inject overwriting properties in the state object.